src/util/projection.ts

```diff
@@ -0,0 +1,45 @@
+import v8 from '../reference/v8.json' with { type: 'json' };
+import {ProjectionPrimitiveT, ProjectionT, ProjectionSpecification, PropertyValueSpecification} from '../types.g';
+
+export class Projection {
+    readonly from: ProjectionPrimitiveT;
+    readonly to: ProjectionPrimitiveT;
+    readonly transition: number;
+    constructor(from: ProjectionPrimitiveT, to: ProjectionPrimitiveT, transition: number){
+        this.from = from;
+        this.to = to;
+        this.transition = transition;
+    }
+
+    toString() {
+        return `["${this.from}", "${this.to}", ${this.transition}]`;
+    }
+}
+
+export function isProjectionConfig(value: unknown): value is ProjectionSpecification {
+    return isProjectionPrimitive(value) || isProjectionValue(value) || isPropertyValueSpecification(value);
+}
+
+export function isPropertyValueSpecification(value: unknown): value is PropertyValueSpecification<ProjectionT> {
+
+    if (['interpolate-projection', 'step', 'literal'].includes(value[0])) {
+        return true
+    }
+    return false
+}
+
+export function isProjectionValue(value: unknown): value is ProjectionT {
+    return Array.isArray(value) &&
+        value.length === 3 &&
+        isProjectionPrimitive(value[0]) &&
+        isProjectionPrimitive(value[1]) &&
+        typeof value[2] === 'number';
+}
+
+export function isProjectionPrimitive(value: unknown): value is ProjectionPrimitiveT {
+    return Object.keys(v8.projectionConfig.type.values.projections).includes(value as string);
```

Comment:
I now see that this is used here, is there an equivalent for this in the style spec?
If not, I would consider hardcoding it here, although I don't like to hardcode stuff, setting it in the v8.json file and parsing it in all kinds of places (think about the "i" in maputnik for example) is not a great experience.
I don't know... this is only a though...

Reply:
Alright, I'll remove the projections/primitives. It won't fly for WKT / +proj anyway, and it'll get out of sync with Native since it's not possible for us to make "sdk-support" on a value level.